src/ecdsapod.rs

```diff
@@ -0,0 +1,546 @@
+        pw.set_biguint_target(&self.signature.s.value, &biguint_from_array(signature.s.0))?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug)]
+struct EcdsaPodVerifyTarget {
+    vds_root: HashOutTarget,
+    id: HashOutTarget,
+    proof: ProofWithPublicInputsTarget<D>,
+}
+
+pub struct EcdsaPodVerifyInput {
+    vds_root: Hash,
+    id: PodId,
+    proof: ProofWithPublicInputs<F, C, D>,
+}
+impl EcdsaPodVerifyTarget {
+    fn add_targets(builder: &mut CircuitBuilder<F, D>, params: &Params) -> Result<Self> {
+        let measure = measure_gates_begin!(builder, "EcdsaPodVerifyTarget");
+
+        // Verify EcdsaVerifyTarget's proof (with verifier_data hardcoded as constant)
+        let (_, circuit_data) = &*ECDSA_VERIFY_DATA;
+        let verifier_data_targ = builder.constant_verifier_data(&circuit_data.verifier_only);
+        let proof_targ = builder.add_virtual_proof_with_pis(&circuit_data.common);
+        builder.verify_proof::<C>(&proof_targ, &verifier_data_targ, &circuit_data.common);
+
+        // calculate id
+        let msg = &proof_targ.public_inputs[0..8];
+        let pk = &proof_targ.public_inputs[8..24];
+        let statements = pub_self_statements_target(builder, params, msg, pk);
+        let id = CalculateIdGadget {
+            params: params.clone(),
+        }
+        .eval(builder, &statements);
+
+        // register the public inputs
+        let vds_root = builder.add_virtual_hash();
+        builder.register_public_inputs(&id.elements);
+        builder.register_public_inputs(&vds_root.elements);
+
+        measure_gates_end!(builder, measure);
+        Ok(EcdsaPodVerifyTarget {
+            vds_root,
+            id,
+            proof: proof_targ,
+        })
+    }
+
+    fn set_targets(&self, pw: &mut PartialWitness<F>, input: &EcdsaPodVerifyInput) -> Result<()> {
+        pw.set_proof_with_pis_target(&self.proof, &input.proof)?;
+        pw.set_hash_target(self.id, HashOut::from_vec(input.id.0.0.to_vec()))?;
+        pw.set_target_arr(&self.vds_root.elements, &input.vds_root.0)?;
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct EcdsaPod {
+    params: Params,
+    id: PodId,
+    msg: Secp256K1Scalar,
+    pk: ECDSAPublicKey<Secp256K1>,
+    proof: Proof,
+    vds_hash: Hash,
+}
+impl middleware::RecursivePod for EcdsaPod {
+    fn verifier_data(&self) -> VerifierOnlyCircuitData<C, D> {
+        let (_, circuit_data) = &*STANDARD_ECDSA_POD_DATA;
+        circuit_data.verifier_data().verifier_only
+    }
+    fn proof(&self) -> Proof {
+        self.proof.clone()
+    }
+    fn vds_root(&self) -> Hash {
+        self.vds_hash
+    }
+}
+
+static STANDARD_ECDSA_POD_DATA: LazyLock<(EcdsaPodVerifyTarget, CircuitData<F, C, D>)> =
+    LazyLock::new(|| build().expect("successful build"));
+
+fn build() -> Result<(EcdsaPodVerifyTarget, CircuitData<F, C, D>)> {
+    let params = &*pod2::backends::plonky2::DEFAULT_PARAMS;
+    let config = CircuitConfig::standard_recursion_config();
```

Comment:
I recommend not creating the config like this and instead using the same config as the `STANDARD_REC_MAIN_POD_CIRCUIT_DATA` so that everything still works if we change the standard recursive MainPod config:
```rust
let rec_circuit_data = &*pod2::backends::plonky2::STANDARD_REC_MAIN_POD_CIRCUIT_DATA;
let config = rec_circuit_data.common.config.clone();
```